### PR TITLE
feat: add `list` classmethod to `FeedbackDataset` via `ArgillaMixin`

### LINKED ISSUE #3413: Add `delete` and `list` methods for `FeedbackDataset` in Argilla via the Python client
Since we can now delete workspaces, we need to tackle two main goals to improve this feature when datasets are linked to the workspace:

- [ ] Support for removing a feedback dataset from the Python client
```python
...
ds = FeedbackDataset.from_argilla(...)
ds.delete()
# or just (since the from_argilla will try to fetch all records locally
FeedbackDataset.delete(dataset_id)
```

- [ ] Fetch the infor about which datasets are linked to the workspace 

_As a first approach, we could return the needed info as part of the delete error response. The name for "old" datasets, and the id for feedback ones_

### Listing and deleting old task datasets
```python
import argilla as rg

ws = rg.Workspace.from_name("ws")

# listing old datasets for a workspace
datasets = rg.list_datasets(workspace=ws)
# or for all workspaces
datasets = rg.list_datasets()

# deleting old datasets
for dataset in datasets:
    rg.delete(name=dataset.name, workspace=datase.workspace)
```

### Listing and deleting new feedback datasets

```python
from argilla.feedback import FeedbackDataset

datasets = FeedbackDataset.list(workspace=ws)
# or all workspaces
datasets = FeedbackDataset.list()
for dataset in datasets:
    dataset.delete()
```

### PR DESCRIPTION
**Pull request by alvarobartt**
# Description

This PR adds the class method `list` in `ArgillaMixin`, also renamed from `ArgillaToFromMixin`. This way, users can call `FeedbackDataset.list()` to list all the `FeedbackDataset` datasets in Argilla.

Additionally, `list` includes the arg `workspace` so that users can also specify the `workspace` that they want to list from, in case those are assigned to more than one workspace.

Closes #3413
Mentioned #3521 #3591

**Type of change**

- [X] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [X] Add integration tests for `FeedbackDataset.list` (producing a list of `RemoteFeedbackDataset`)

**Checklist**

- [X] I added relevant documentation
- [X] follows the style guidelines of this project
- [X] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [x] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/client/feedback/dataset/mixins.py [hunk 1]
```diff
@@ -332,3 +332,46 @@ def from_argilla(
...
+    def list(cls: Type["FeedbackDataset"], workspace: Optional[str] = None) -> List[RemoteFeedbackDataset]:
+        """Lists the `FeedbackDataset`s pushed to Argilla.
+
+        Note that you may need to `rg.init(...)` with your Argilla credentials before
+        calling this function, otherwise, the default http://localhost:6900 will be used,
+        which will fail if Argilla is not deployed locally.
+
+        Args:
+            workspace: the workspace where to list the datasets from. Defaults to `None`
+                which means that no workspace filtering will be applied.
```
Comment (gabrielmbmb) [inline src/argilla/client/feedback/dataset/mixins.py]: With Google Docstring style I always prefer to have the "Defaults to ..." at the end of the arg description. Feel free to ignore this.

```suggestion
            workspace: the workspace where to list the datasets from. If not provided,
                then the workspace filtering won't be applied. Defaults to `None`.
```

File: src/argilla/client/feedback/dataset/mixins.py [hunk 2]
```diff
@@ -332,3 +332,46 @@ def from_argilla(
...
+        if workspace is not None:
+            workspace = Workspace.from_name(workspace)
+
+        # TODO(alvarobartt or gabrielmbmb): add `workspace_id` in `GET /api/v1/datasets`
+        # and in `GET /api/v1/me/datasets` to filter by workspace
+        try:
+            datasets = datasets_api_v1.list_datasets(client=httpx_client).parsed
+        except Exception as e:
+            raise Exception(
+                "Failed while listing the `FeedbackDataset` datasets in Argilla with" f" exception: {e}"
```
Comment (gabrielmbmb) [inline src/argilla/client/feedback/dataset/mixins.py]:
```suggestion
                f"Failed while listing the `FeedbackDataset` datasets in Argilla with exception: {e}"
```

File: src/argilla/client/feedback/dataset/mixins.py [hunk 3]
```diff
@@ -332,3 +332,46 @@ def from_argilla(
...
+
+        if workspace is not None:
+            workspace = Workspace.from_name(workspace)
+
+        # TODO(alvarobartt or gabrielmbmb): add `workspace_id` in `GET /api/v1/datasets`
+        # and in `GET /api/v1/me/datasets` to filter by workspace
+        try:
+            datasets = datasets_api_v1.list_datasets(client=httpx_client).parsed
+        except Exception as e:
+            raise Exception(
```
Comment (gabrielmbmb) [inline src/argilla/client/feedback/dataset/mixins.py]: maybe it's better to use `RuntimeError`?

File: src/argilla/client/feedback/dataset/mixins.py [hunk 4]
```diff
@@ -332,3 +332,46 @@ def from_argilla(
...
+            RemoteFeedbackDataset(
+                client=httpx_client,
+                id=dataset.id,
+                name=dataset.name,
+                workspace=workspace if workspace is not None else Workspace.from_id(dataset.workspace_id),
+                fields=cls.__get_fields(client=httpx_client, id=dataset.id),
+                questions=cls.__get_questions(client=httpx_client, id=dataset.id),
+                guidelines=dataset.guidelines or None,
+            )
+            for dataset in datasets
```
Comment (gabrielmbmb) [inline src/argilla/client/feedback/dataset/mixins.py:376]: in the meantime, we can apply the filtering in the client, right?

```suggestion
            for dataset in datasets if workspace is None or dataset.workspace_id == workspace.id
```

**Review by gabrielmbmb — COMMENTED** [review]
LGTM! I think that we can filter the datasets in the client until we add the filter in the API, wdyt?

File: src/argilla/client/feedback/dataset/mixins.py (hunk 3 above)
Comment (alvarobartt) [reply src/argilla/client/feedback/dataset/mixins.py]:
```suggestion
            raise RuntimeError(
```

**Review by gabrielmbmb — APPROVED** [review]

**Pull request merged**